src/Learning/Parameters.jl

```diff
@@ -173,6 +173,51 @@ function backpropagate_tree!(Δ::AbstractMatrix{<:Real}, C::Vector{Node}, V::Abs
   return nothing
 end
 
+```
+  Marginalized version of backpropagate_tree.
+```
+function norm_backpropagate_tree!(marg_Δ::AbstractVector{<:Real}, C::Vector{Node}, marg_V::AbstractVector{<:Real})
+  marg_Δ .= 0.0
```

Comment:
Not all of the matrix needs to be set to zero. See https://github.com/RenatoGeh/RPCircuits.jl/blob/a4cfef7c92c324aa4252f8c7ed834d1d6ac4e4af/src/Learning/Parameters.jl#L66 for more information. We only need to set to zero entries which are either incremented throughout the loop or belong to the root.